dvc/repo/__init__.py

```diff
@@ -184,10 +211,10 @@ def __init__(
                 friendly=True,
             )
 
-            # NOTE: storing state and link_state in the repository itself to
-            # avoid any possible state corruption in 'shared cache dir'
-            # scenario.
-            self.state = State(self.root_dir, self.tmp_dir, self.dvcignore)
+            state_tmp_dir = self._get_subsystem_tmp_dir(
+                "state", self.root_dir, self.tmp_dir, self.config
```

Comment:
In both use cases the only actual argument we provide is `state` or `index`, rest is repo class variables. Do we need to explicitly pass them?

Reply:
Good point, we don't.